slack-plugin/src/main/kotlin/slack/gradle/lint/LintTasks.kt

```diff
@@ -87,139 +255,196 @@ internal object LintTasks {
+      checkTestSources = slackProperties.lintCheckTestSources
+
+      textReport = true
+      xmlReport = true
+      htmlReport = true
+      sarifReport = true
+
+      // Format text output for convenience.
+      explainIssues = true
+      noLines = false
+      quiet = true
+
+      // We run lint on each library, so we don't want transitive checking of each dependency
+      checkDependencies = false
+
+      fatal.add("VisibleForTests")
+
+      if (isMultiplatform) {
+        // Disable classfile-based checks because lint cannot find the class files for
+        // multiplatform projects and `SourceSet.java.classesDirectory` is not configurable.
+        // This is not ideal, but it's better than having no lint checks at all.
+        disable.add("LintError")
+      }
+
+      // Disable dependency checks that suggest to change them. We want libraries to be
+      // intentional with their dependency version bumps.
+      disable.add("KtxExtensionAvailable")
+      disable.add("GradleDependency")
+
+      // These store qualified gradle caches in their paths and always change in baselines
+      disable += "ObsoleteLintCustomCheck"
+
+      // Explicitly disable StopShip check (see b/244617216)
+      disable.add("StopShip")
+
+      fatal.add("Assert")
+      fatal.add("NewApi")
+      fatal.add("ObsoleteSdkInt")
+      fatal.add("NoHardKeywords")
+      fatal.add("UnusedResources")
+      fatal.add("KotlinPropertyAccess")
+      fatal.add("LambdaLast")
+
+      // Too many Kotlin features require synthetic accessors - we want to rely on R8 to
+      // remove these accessors
+      disable.add("SyntheticAccessor")
 
-    slackProperties.ciLintVariants?.let { variants ->
-      ciLintTask.configure {
-        // Even if the task isn't created yet, we can do this by name alone and it will resolve at
-        // task configuration time.
-        variants.splitToSequence(',').forEach { variant ->
-          logger.debug("Using variant $variant for ciLint task")
-          val lintTaskName = "lint${variant.capitalizeUS()}"
-          dependsOn(lintTaskName)
+      // Only check for missing translations in finalized (beta and later) modules.
+      fatal.add("MissingTranslation")
+
+      if (androidExtensionNullable is ApplicationAndroidComponentsExtension) {
+        checkDependencies = true
+      }
+
+      androidSdkVersions?.let { sdkVersions ->
+        if (sdkVersions.minSdk >= 28) {
+          // Lint doesn't understand AppComponentFactory
+          // https://issuetracker.google.com/issues/243267012
+          disable += "Instantiatable"
         }
       }
-      return // nothing else to do!
+
+      lintConfig = project.rootProject.layout.projectDirectory.file("config/lint/lint.xml").asFile
+
+      baseline =
+        project.objects
+          .fileProperty()
+          .fileValue(File(project.projectDir, slackProperties.lintBaselineFileName))
+          .get()
+          .asFile
```

Comment:
this seems like a byzantine way to get a file reference in the current project 😅 

why doesn't something like this work?
```kotlin
project.layout.projectDirectory.file(slackProperties.lintBaselineFileName).get().asFile
```
I mean... not that that's so much less byzantine...

Reply:
I thought this was doing something for snapshotting contents only but after reading the docs I am wrong. [f12545b](https://github.com/slackhq/slack-gradle-plugin/pull/612/commits/f12545b5009deb88efa3ef717a50e84366c569fb)